airbyte-integrations/connectors/destination-milvus/destination_milvus/indexer.py

```diff
@@ -0,0 +1,106 @@
+        super().__init__(config)
+        self.embedder_dimensions = embedder_dimensions
+
+    def _connect(self):
+        connections.connect(
+            uri=self.config.host,
+            db_name=self.config.db if self.config.db else "",
+            user=self.config.auth.username if self.config.auth.mode == "username_password" else "",
+            password=self.config.auth.password if self.config.auth.mode == "username_password" else "",
+            token=self.config.auth.token if self.config.auth.mode == "token" else "",
+        )
+
+    def _create_client(self):
+        # Run connect in a separate process as it will hang if the token is invalid.
+        proc = Process(target=self._connect)
+        proc.start()
+        proc.join(5)
+        if proc.is_alive():
+            # If the process is still alive after 5 seconds, terminate it and raise an exception
+            proc.terminate()
+            proc.join()
+            raise Exception("Connection timed out, check your host and credentials")
+
+        # If the process exited within 5 seconds, it's safe to connect on the main process to execute the command
+        self._connect()
+
+        self._collection = Collection(self.config.collection)
+        self._collection.load()
+
+    def check(self) -> Optional[str]:
+        deployment_mode = os.environ.get("DEPLOYMENT_MODE", "")
+        if deployment_mode.casefold() == CLOUD_DEPLOYMENT_MODE and not self._uses_https():
+            return "Host must start with https://"
+        try:
+            self._create_client()
+
+            description = self._collection.describe()
+            if not description["auto_id"]:
+                return "Only collections with auto_id are supported"
+            vector_field = next((field for field in description["fields"] if field["name"] == self.config.vector_field), None)
+            if vector_field is None:
+                return f"Vector field {self.config.vector_field} not found"
+            if vector_field["type"] != DataType.FLOAT_VECTOR:
+                return f"Vector field {self.config.vector_field} is not a vector"
+            if vector_field["params"]["dim"] != self.embedder_dimensions:
+                return f"Vector field {self.config.vector_field} is not a {self.embedder_dimensions}-dimensional vector"
+        except DescribeCollectionException:
+            return f"Collection {self.config.collection} does not exist"
+        except Exception as e:
+            return format_exception(e)
+        return None
+
+    def _uses_https(self) -> bool:
+        return self.config.host.startswith("https://")
+
+    def pre_sync(self, catalog: ConfiguredAirbyteCatalog) -> None:
+        self._create_client()
+        for stream in catalog.streams:
+            if stream.destination_sync_mode == DestinationSyncMode.overwrite:
+                self._delete_for_filter(f'{METADATA_STREAM_FIELD} == "{stream.stream.name}"')
```

Comment:
Just a question, do these sync run fast enough to where clearing everything out isn't an issue for downstream applications that may be using this?

I know in destinations v2 we recently moved to doing an old table drop after the data was ready so the downtime is minimal (internally records go to a temp table, and then we drop the old table and rename the temp table to the real name when ready). We might not have to do something as involved, but am wondering if you think there's something to improve here with how we handle the overwrites. I think specially in rETL-like cases this is more important.

Reply:
That's a very good point - for overwrite-syncs there's definitely the potential for issues as it will first delete everything, then add stuff back. IMHO it's not required for the v1, but definitely a feature we should add later on. Append/append+dedup should be fine in most cases (if referential integrity among records doesn't matter which is probably common for vector dbs), but it might still be helpful.